app/static/js/cluster.js

```diff
@@ -54,6 +54,10 @@ fetch('js/data.json', {mode: 'no-cors'})
 
     elements: data
   });
+  var png64 = cy.png();
+
+  $('<a id="png" download>Download Image!</a>').insertBefore('#cy');
+    $('#png').attr('href', png64);
 
   var layout = cy.layout( options );
```

Comment:
The image that downloads doesn't have a circular layout- could you move the `var layout = cy.layout( options );` line above the image lines?